blob/driver/driver.go

```diff
@@ -38,13 +38,20 @@ type Error interface {
 	BlobError() ErrorKind
 }
 
+// NoAs can be used as an As function that always returns false.
+var NoAs = func(interface{}) bool { return false }
+
 // Reader reads an object from the blob.
 type Reader interface {
 	io.ReadCloser
 
 	// Attributes returns a subset of attributes about the blob.
-	// Use Bucket.Attributes to get the full set.
 	Attributes() ReaderAttributes
+
+	// As allows providers to expose provider-specific types.
+	// See https://github.com/google/go-cloud/blob/master/internal/docs/design.md#escape-hatches
+	// for more details.
+	As(interface{}) bool
```

Comment:
Does this need to be a required method? If it was introduced in a new interface and dynamically tested for, then it would not be a breaking change.

Reply:
It could be done that way, but given that we don't need to be compatible at this point, I think it's cleaner to just add it.